cdap-explore/src/main/java/co/cask/cdap/hive/stream/StreamRecordReader.java

```diff
@@ -45,12 +52,22 @@
   private final StreamDataFileReader reader;
   private final StreamInputSplit inputSplit;
   private final ReadFilter readFilter;
+  private final Principal principal;
+  private final StreamId streamId;
+  private final AuthorizationEnforcer authorizationEnforcer;
 
   StreamRecordReader(InputSplit split, JobConf conf) throws IOException {
+    ContextManager.Context context = ContextManager.getContext(conf);
     this.inputSplit = (StreamInputSplit) split;
     this.events = Lists.newArrayListWithCapacity(1);
     this.reader = createReader(FileSystem.get(conf), inputSplit);
     this.readFilter = new TimeRangeReadFilter(inputSplit.getStartTime(), inputSplit.getEndTime());
+    Preconditions.checkArgument(context != null, "ContextManager should not be null in Hive MR job.");
```

Comment:
this may not be a MR job, right?
The RecordReader might be used from other engines also?